### PR TITLE
[Metrics] Add a few Websocket metrics

### PR DESCRIPTION
**Pull request by quantranhong1999**
- Opening connections count
- Requests throughput

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala [hunk 1]
```diff
@@ -79,7 +80,12 @@ class WebSocketRoutes @Inject() (@Named(InjectionKeys.RFC_8621) val authenticato
                                  emailChangeRepository: EmailChangeRepository,
                                  pushSerializer: PushSerializer,
                                  typeStateFactory: TypeStateFactory,
-                                 delegationStore: DelegationStore) extends JMAPRoutes {
+                                 delegationStore: DelegationStore,
+                                 metricFactory: MetricFactory,
+                                 gaugeRegistry: GaugeRegistry) extends JMAPRoutes {
+  private val openingConnectionsMetric: Metric = metricFactory.generate("jmap_websocket_opening_connections")
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala:86]:
```suggestion
  private val openingConnectionsMetric: Metric = metricFactory.generate("jmap_websocket_opened_connections")
```

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala [hunk 2]
```diff
@@ -113,6 +120,10 @@ class WebSocketRoutes @Inject() (@Named(InjectionKeys.RFC_8621) val authenticato
         frame.content().readBytes(bytes)
         new String(bytes, StandardCharsets.UTF_8)
       })
+      .doOnNext(_ => {
+        requestCountMetric.increment()
+        requestThroughputGauge.setValue(requestCountMetric.meanRate().intValue)
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala:125]: Kill that gauge, it won't be performant. And we can do a derivative for that in grafana.

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala (hunk 1 above)
Comment (quantranhong1999) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/WebSocketRoutes.scala]: Do you mean opened connections of all time, or just opened connections at the moment?

**Review by chibenwa — APPROVED** [review]
We can do IMO the reactor.netty http metrics separately...

**Review by Arsnael — APPROVED** [review]

**Comment by quantranhong1999** [comment]
Squashed.

**Pull request merged**